### PR TITLE
Feature/display lessons

### LINKED ISSUE #66: Lessons component
- [x] Should see a list of all lessons
- [x] Should be able to click on the lesson and see lesson details

### PR DESCRIPTION
**Pull request by Melizzo**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures
## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe): 
## What is the current behavior?

- simple display lesson page has been set
## What is the new behavior?

-Displaying all lessons
-adding lessonCard to place card on the DOM
-adding modal for future iterations of lesson statistics
## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

closes #66 

Screen Shot 2020-09-12 at 6 41 32 PM
Screen Shot 2020-09-12 at 6 41 54 PM

### CODE REVIEW

File: src/thunks/__tests__/getStudents.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ describe('getStudents', () => {
           "last_name": "newStudent2Last" 
       }   
     ]   
-    mockUrl = `https://gumberoo-backend.herokuapp.com/api/v1/teachers/${mockTeacherId}/students`
+    mockUrl = `https://gumberoo-backend.herokuapp.com/api/v1/teachers/${mockTeacherId}/students/`
```
Comment (Billwilke42) [inline src/thunks/__tests__/getStudents.js:27]: thanks for adding these back!

File: src/TeacherLessons/TeacherLessons.js [hunk 2]
```diff
@@ -1,45 +1,77 @@
...
-    <main className='teacher-lessons'>
+    <main className="teacher-lessons">
       <h1>Lessons</h1>
-
       {lessons.length ? (
-        <LessonCard /> ) : (
-          <p className="no-lessons-message">Please click on the "Create a Lesson" Tab to create a lessson.</p>
-        )
-      }
+        (lessonCard = lessons.map((lesson) => {
```
Comment (Billwilke42) [inline src/TeacherLessons/TeacherLessons.js:50]: excited to check this out!

File: src/TeacherDashboard/TeacherDashboard.test.js [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
-import { render, fireEvent } from '@testing-library/react';
+import { render } from '@testing-library/react';
+import { createStore } from 'redux'
 import { Provider } from 'react-redux';
 import TeacherDashboard from './TeacherDashboard';
 import { BrowserRouter } from 'react-router-dom';
+import { rootReducer } from '../reducers';
+import { addLesson } from '../mockData/mockData'
+
+const store = createStore(rootReducer, {
```
Comment (Billwilke42) [inline src/TeacherDashboard/TeacherDashboard.test.js:11]: nice work creating the store!

**Review by Billwilke42 — APPROVED** [review]
Great work, I really hope moving the redirect file fixes our routes on netifly, so close!

File: src/CreateLesson/CreateLesson.js [hunk 4]
```diff
@@ -10,7 +10,6 @@ const CreateLesson = () => {
   const deleteQuestion = (e) => {
     //eslint-disable-next-line
```
Comment (KyleWong2510) [inline src/CreateLesson/CreateLesson.js:11]: we could likely use the + to make that a number and not use those disable lines here.

File: src/TeacherLessons/TeacherLessons.js [hunk 5]
```diff
@@ -1,45 +1,77 @@
...
 // import { bindActionCreators } from "redux";
 
-const TeacherLessons = ({students, lessons}) => {
-  // eslint-disable-next-line
-  const [error, setError] = useState('')
+const TeacherLessons = ({ students, lessons }) => {
+  const [foundLesson, setFoundLesson] = useState({})
+  const [isViewingLessonDetails, toggleLessonDetails] = useState(false)
+
+  const findLesson = (e) => {
```
Comment (KyleWong2510) [inline src/TeacherLessons/TeacherLessons.js:14]: Nice job working through this one!

**Review by KyleWong2510 — APPROVED** [review]
Nice work on this, Melizzo!

**Pull request merged**